service/controller/resource/app/desired.go

```diff
@@ -203,17 +206,48 @@ func (r *Resource) newApp(clusterConfig v1alpha1.ClusterGuestConfig, appSpec key
 	}
 }
 
-func (r *Resource) newAppSpecs() []key.AppSpec {
-	switch r.provider {
-	case "aws":
-		return append(key.CommonAppSpecs(), awskey.AppSpecs()...)
-	case "azure":
-		return append(key.CommonAppSpecs(), azurekey.AppSpecs()...)
-	case "kvm":
-		return append(key.CommonAppSpecs(), kvmkey.AppSpecs()...)
-	default:
-		return key.CommonAppSpecs()
+func (r *Resource) newAppSpecs(ctx context.Context, cr v1alpha1.ClusterGuestConfig) ([]key.AppSpec, error) {
+	req := searcher.Request{
+		ReleaseVersion: cr.ReleaseVersion,
+	}
+
+	res, err := r.clusterClient.Release.Searcher.Search(ctx, req)
```

Comment:
`legacy` does not contain `operatorversion` resource as master branch. So `app` resource would fetch release info by itself.